geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/FilterValues.scala

```diff
@@ -0,0 +1,61 @@
+/***********************************************************************
+ * Copyright (c) 2013-2017 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.filter
+
+import scala.collection.GenTraversableOnce
+
+/**
+  * Holds values extracted from a filter. Values may be empty, in which case nothing was extracted from
+  * the filter. May be marked as 'disjoint', which means that mutually exclusive values were extracted
+  * from the filter. This may be checked to short-circuit queries that will not result in any hits.
+  *
+  * @param values values extracted from the filter. If nothing was extracted, will be empty
+  * @param precise values exactly match the filter, or may return false positives
+  * @param disjoint mutually exclusive values were extracted, e.g. 'a < 1 && a > 2'
+  * @tparam T type parameter
+  */
+case class FilterValues[T](values: Seq[T], precise: Boolean = true, disjoint: Boolean = false) {
```

Comment:
(Idly) Should this implement filterMonadic or whatever?

Reply:
is there an interface already, or you mean we should create one?

Reply:
http://www.scala-lang.org/api/2.11.8/index.html#scala.collection.generic.FilterMonadic

Using that might allow for use of Scala's synatical sugar around monads.

Reply:
interesting, i'll check it out

Reply:
i got lost down a rabbit hole of CanBuildFrom, etc - maybe I'll revisit later but for now it's too complicated i think